src/Microsoft.ML.Data/DataView/DataViewExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.ML.Data
+{
+    public static class DataViewExtensions
```

Comment:
>DataViewExtensions [](start = 24, length = 18)

Whoops, this class was made public.

Reply:
Did you add this because you thought this is what I meant by a utility? Or was there some other reason? If it was for my benefit, I actually meant (1) something that works purely over `Schema.Column`. In any case, I'm not sure this is a good idea. I think operating over `string` and `int` directly is dangerous and gives up a lot of error checking capability that is roughly 75% of my own motivation for wanting this change. Also whatever "shims" we have, even if necessary, ought to be completely temporary, and ought to work in the *opposite* direction. (That is, I accept that *internally* we might have some things that work directly over `int` indices, so mapping from `Schema.Column` to `int` based structures might be fine. Doing the opposite though, is going in the wrong direction.)

---
In reply to: [245080660](https://github.com/dotnet/machinelearning/pull/1975#discussion_r245080660) [](ancestors = 245080660)

Reply:
you did suggest that its signature involves the Schema.Column and looks like this:

<code>GetRowCursor(IEnumerable<Schema.Column> colsNeeded)</code>

But since in most places the code looked like this:

1- Getting the indices of the columns needed, 
2- subset those indices from the schema
3- pass those columns to this method

i compacted the above by creating those methods that take in directly column names or indices. 

To enable that validation that you are suggesting, and encapsulating the columns sub setting from the schema here, i think the following should work, let me know if something is a miss:

1- Internalize this class, and its methods. 
2- keep the method signatures as they are, to centralize the way the schema gets accessed, and the validation
3- before accessing the columns on the schema, here, validate they exist and throw. 


---
In reply to: [245082390](https://github.com/dotnet/machinelearning/pull/1975#discussion_r245082390) [](ancestors = 245082390,245080660)